### DISCUSSION TITLE
can i pass value from dialog to Serenity.EntityGrid editor

### DISCUSSION THREAD

**Discussion by clarklin7475** [serenity-is/Serenity#6131] [Help] [answered]
**dialog.ts code:**

namespace SystemMenuGroupAuthority.SystemDB {

    @Serenity.Decorators.registerClass()
    export class SystemDialog extends Serenity.EntityDialog<SystemRow, any> {
        protected getFormKey() { return SystemForm.formKey; }
        protected getIdProperty() { return SystemRow.idProperty; }
        protected getLocalTextPrefix() { return SystemRow.localTextPrefix; }
        protected getNameProperty() { return SystemRow.nameProperty; }
        protected getService() { return SystemService.baseUrl; }
        protected getDeletePermission() { return SystemRow.deletePermission; }
        protected getInsertPermission() { return SystemRow.insertPermission; }
        protected getUpdatePermission() { return SystemRow.updatePermission; }

        protected form = new SystemForm(this.idPrefix);
    }

}

**editor.ts code:**

namespace SystemMenuGroupAuthority.SystemDB {

    @Serenity.Decorators.registerEditor()
    export class SystemMenuTreeGrid extends Serenity.EntityGrid<SystemMenuRow, any> {//Extensions.GridEditorBase
        protected getColumnsKey() { return 'SystemDB.SystemMenu'; }
        protected getDialogType() { return SystemMenuDialog; }
        protected getIdProperty() { return SystemMenuRow.idProperty; }
        //protected getInsertPermission() { return SystemMenuRow.insertPermission; }
        //protected getLocalTextPrefix() { return SystemMenuRow.localTextPrefix; }
        protected getService() { return SystemMenuService.baseUrl; }
        private treeMixin: Serenity.TreeGridMixin<SystemDB.SystemMenuRow>;
        //constructor(container: JQuery) {
        //    super(container);
        //}
        public SystemId: number;

        constructor(container: JQuery) {
            super(container);

            var a = this;
            this.treeMixin = new Serenity.TreeGridMixin({
                grid: this,
          
                // bring tree items initially collapsed
                initialCollapse: () => true,
                // which column to place tree toggle / expand/collapse button
                toggleField: SystemDB.SystemMenuRow.Fields.Name,
                getParentId: x => {
                
                    return x.ParentCode;
                }

            });
        }
     
        protected getAddButtonCaption() {
            return "新增";
        }

        protected onViewSubmit(): boolean {
            if (!super.onViewSubmit()) {
                return false;
            }
            let request = this.view.params as Serenity.ListRequest;
            request.Criteria = Serenity.Criteria.and(request.Criteria, [['SystemId'], '=', this.SystemId]);// 
            return true;
        }

    
    }
}

there has SystemId in Dialog, can i get SystemId in protected onViewSubmit() function?

**Comment by wezmag** [comment, marked as answer]
I think you can find the answer in the official Movie Database Tutorial. 
It's in the "Filtering Movies for the Person" section.
https://serenity.is/docs/tutorials/movies/listing_movies_in_person_dialog

**Reply by clarklin7475** [reply]
thank you
